chain-abci/src/app/mod.rs

```diff
@@ -130,27 +130,27 @@ impl<T: EnclaveProxy> abci::Application for ChainNodeApp<T> {
             ),
         };
 
-        {
-            let last_state = self
+        let last_state = self
             .last_state
             .as_mut()
             .expect("executing begin block, but no app state stored (i.e. no initchain or recovery was executed)");
 
-            last_state.block_time = block_time;
-
-            if block_height > 1 {
-                if let Some(last_commit_info) = req.last_commit_info.as_ref() {
-                    // liveness will always be updated for previous block, i.e., `block_height - 1`
-                    update_validator_liveness(last_state, block_height - 1, last_commit_info);
-                } else {
-                    panic!(
-                        "No last commit info in begin block request for height: {}",
-                        block_height
-                    );
-                }
+        last_state.block_time = block_time;
+
+        if block_height > 1 {
+            if let Some(last_commit_info) = req.last_commit_info.as_ref() {
+                // liveness will always be updated for previous block, i.e., `block_height - 1`
+                update_validator_liveness(last_state, block_height - 1, last_commit_info);
+            } else {
+                panic!(
+                    "No last commit info in begin block request for height: {}",
+                    block_height
+                );
             }
         }
 
+        let mut accounts_to_jail = Vec::new();
```

Comment:
should this be a vector and not a set -- but I guess jailing is idempotent, so it shouldn't matter if the validator was both non-live and byzantine?

Reply:
That is correct. Jailing is idempotent. In this case, `Vec` is preferred as compare to `BTreeSet` because it is contiguous memory and requires zero pointer jumps for iteration (better performance while iterating).